Experimental/DistributedAuthoritySample/Assets/Scripts/Player/AvatarTransform.cs

```diff
@@ -0,0 +1,272 @@
+            m_Rigidbody.linearVelocity = Vector3.zero;
+        }
+
+        void Update()
+        {
+            if (!IsSpawned || !HasAuthority)
+            {
+                return;
+            }
+
+            m_Movement = new Vector3(m_AvatarInputs.move.x, 0, m_AvatarInputs.move.y).normalized;
+
+            // Handle rotation based on input direction
+            if (m_Movement.magnitude >= 0.1f)
+            {
+                var targetAngle = Mathf.Atan2(m_Movement.x, m_Movement.z) * Mathf.Rad2Deg;
+                var targetRotation = Quaternion.Euler(0, targetAngle, 0);
+                transform.rotation = Quaternion.Lerp(transform.rotation, targetRotation, Time.deltaTime * m_RotationSpeed);
+            }
+
+            if (IsGrounded() && m_AvatarInputs.jump)
+            {
+                m_Jump = true;
+                m_AvatarInputs.jump = false;
+            }
+        }
+
+        void ApplyMovement()
+        {
+            if (Mathf.Approximately(m_Movement.magnitude, 0f))
+            {
+                return;
+            }
+
+            var velocity = m_Rigidbody.linearVelocity;
+            var desiredVelocity = m_Movement * (m_AvatarInputs.sprint ? m_SprintSpeed : m_WalkSpeed);
+            var targetVelocity = new Vector3(desiredVelocity.x, velocity.y, desiredVelocity.z);
+            var velocityChange = targetVelocity - velocity;
+
+            if (m_IsGrounded)
+            {
+                // Apply force proportional to acceleration while grounded
+                var force = velocityChange * m_Acceleration;
+                m_Rigidbody.AddForce(force, ForceMode.Acceleration);
+            }
+            else
+            {
+                // Apply reduced force in the air for air control
+                var force = velocityChange * (m_Acceleration * m_AirControlFactor);
+                m_Rigidbody.AddForce(force, ForceMode.Acceleration);
+            }
+        }
+
+        void ApplyJump()
+        {
+            if (m_IsGrounded && m_Jump)
+            {
+                m_Rigidbody.AddForce(Vector3.up * m_JumpImpusle, ForceMode.Impulse);
+                m_Jump = false;
+            }
+        }
+
+        void UpdateGroundedStatus()
+        {
+            m_IsGrounded = IsGrounded();
+        }
+
+        bool IsGrounded()
+        {
+            // Perform a raycast to check if the character is grounded
+            m_Ray.origin = m_Rigidbody.worldCenterOfMass;
+            m_Ray.direction = Vector3.down;
+            return Physics.RaycastNonAlloc(m_Ray, m_RaycastHits, m_GroundCheckDistance) > 0;
+        }
+
+        void FixedUpdate()
+        {
+            if (!IsSpawned || !HasAuthority || m_Rigidbody != null && m_Rigidbody.isKinematic)
+            {
+                return;
+            }
+
+            UpdateGroundedStatus();
+
+            ApplyMovement();
+
+            ApplyJump();
+
+            ApplyDrag();
+
+            ApplyCustomGravity();
```

Comment:
nitpick: lot of uneeded space here

Reply:
Controversial, but I do like the spacing here!